sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -179,6 +185,7 @@ trait ProgressReporter extends Logging {
       name = name,
       timestamp = formatTimestamp(currentTriggerStartTimestamp),
       batchId = currentBatchId,
+      batchDuration = currentTriggerEndTimestamp - currentTriggerStartTimestamp,
```

Comment:
This can be calculated at the early of `finishTrigger()` and reused.

Reply:
fixed